File: 014.NN_UIViewAnimationPractice/NN_UIViewAnimationPractice/UIViewAnimationPractice/Controllers/NNAlphaAnimationViewController.m
//
//  NNAlphaAnimationViewController.m
//  NN_UIViewAnimationPractice
//
//  Created by 刘朋坤 on 2018/8/11.
//  Copyright © 2018年 刘朋坤. All rights reserved.
//

#import "NNAlphaAnimationViewController.h"

@interface NNAlphaAnimationViewController ()

@end

@implementation NNAlphaAnimationViewController

- (NSArray *)buttonTitleArray {
    return @[@"Alpha"];
}

#pragma mark - 处理事件区域
- (void)buttonClick:(UIButton *)sender {
    [self changeAlpha];
}

- (void)changeAlpha {
    [UIView animateWithDuration:1.5 animations:^{
        self.nnView.headerImage.alpha = 0.2;
    } completion:^(BOOL finished) {
        [UIView animateWithDuration:1.5 animations:^{
            self.nnView.headerImage.alpha = 1;
        }];
    }];
}

@end

File: 017.NN_UIViewAnimationPractice_swift/NN_UIViewAnimationPractice_swift/UIViewAnimationPractice/Controllers/NNAlphaAnimationViewController.swift
//
//  NNAlphaAnimationViewController.swift
//  NN_UIViewAnimationPractice
//
//  Created by 刘朋坤 on 2019/1/5.
//  Copyright © 2019 刘朋坤. All rights reserved.
//

import UIKit

class NNAlphaAnimationViewController: NNBaseViewController {

    // 按钮点击事件
    override func buttonClick(index: NSInteger) {
        UIView.animate(withDuration: 1.5, animations: {
            self.nnView.headerImage.alpha = 0.2
        }) { (finished) in
            self.nnView.headerImage.alpha = 1
        }
    }

    // 数组元素
    override func buttonTitleArrayMethod() -> NSArray {
        return ["Alpha"]
    }
    
}
